air-quality-backend/src/air_quality/etl/forecast/forecast_data.py

```diff
@@ -80,18 +110,77 @@ def __init__(self, single_level_data: xr.Dataset, multi_level_data: xr.Dataset):
     _cached_pressure = None
     _cached_temperature = None
 
-    def _get_data_set(self, pollutant_type: PollutantType) -> xr.Dataset:
-        if is_single_level(pollutant_type):
+    def _get_data_set(self, forecast_data_type: ForecastDataType) -> xr.Dataset:
+        if is_single_level(forecast_data_type):
             return self._single_level_data
         else:
             return self._multi_level_data
 
+    def enrich_in_situ_measurements(
+            self,
+            in_situ_measurements: list[InSituMeasurement],
+            required_data: list[ForecastDataType]
+    ) -> list[tuple[InSituMeasurement, dict[ForecastDataType: float]]]:
+        if len(in_situ_measurements) == 0:
+            return []
+
+        latitudes = []
+        longitudes = []
+        valid_times = []
+        for in_situ_measurement in in_situ_measurements:
+            longitude = in_situ_measurement["location"]["coordinates"][0]
+            latitude = in_situ_measurement["location"]["coordinates"][1]
+            valid_time = in_situ_measurement["measurement_date"].timestamp()
+
+            if latitude not in latitudes:
+                latitudes.append(latitude)
+            if longitude not in longitudes:
+                longitudes.append(longitude)
+            if valid_time not in valid_times:
+                valid_times.append(valid_time)
```

Comment:
I'm not sure about this. What would happen if we had these cities:
London, (90, 90)
Near London, (90, 45)

I think we'd end up with:
latitides: [90]
longitudes: [90, 45]

But the function below would expect the list sizes to be the same I think?

Reply:
Oh that's a good point, I'll update and add a unit test to cover as well

Reply:
Actually this one appears to be ok, as I'm already casting these on different dimensions when interpolating. I've added two unit tests to showcase it though, just give me a shout if you think it's still an issue.